### PR TITLE
added students array and validator

### PR DESCRIPTION
**Pull request by Tayef-Shah**
### Notion ticket link

[Add Students to Enrolment Model](https://www.notion.so/uwblueprintexecs/Add-students-to-the-enrolment-model-4548fa4bb9744c04a535f9ecf3b1245e)

### Implementation description

- Added students (array of student IDs) to enrolment
- Added validator to make sure the students belong to the same family as the enrolment
- Added test case for the validator

### Steps to test

1. run tests on enrolments

### What should reviewers focus on?

- code quality
- test completeness

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: enrolments/tests/test_validators.py [hunk 1]
```diff
@@ -76,6 +77,43 @@ def test_validate_enrolments__invalid(self, mock_validate):
...
+            role=Student.CHILD,
+            family=self.family,
+        )
+        child2 = Student.objects.create(
+            first_name="Yohan Wilkshire",
+            last_name="McDonald",
+            role=Student.CHILD,
+            family=self.family,
+        )
+        student_ids = list(map((lambda student: student.id), [parent, child1, child2]))
```
Comment (brittanylau) [inline enrolments/tests/test_validators.py]: would this be lil more concise? `[student.id for student in [parent, child1, child2]`

File: enrolments/tests/test_validators.py [hunk 2]
```diff
@@ -76,6 +77,43 @@ def test_validate_enrolments__invalid(self, mock_validate):
...
+            students=student_ids,
+        )
+
+        self.assertRaises(
+            ValidationError, validate_students_in_enrolment, enrolment, []
+        )
+        self.assertRaises(
+            ValidationError, validate_students_in_enrolment, enrolment, student_ids[1:]
+        )
+        self.assertEqual(enrolment.students, student_ids)
```
Comment (brittanylau) [inline enrolments/tests/test_validators.py]: I think this tests the underlying Django logic more so than your code, so we can remove this line :)

**Review by brittanylau — COMMENTED** [review]
overall test coverage & code quality looks good!

I think there was a miscommunication about the logic – the goal of this task is to enable GSL to enrol a *subset* of the family's members in a session. the main use case is when families return for multiple sessions – when children reach a certain age, they're no longer eligible for the program and won't be able to enrol.

as such, it's actually valid if not every single student is present! we just need to make sure that every id in `enrolment.students` is also in `enrolment.family.students`

lmk if you have any questions about this! 😄

File: enrolments/tests/test_validators.py (hunk 1 above)
Comment (Tayef-Shah) [reply enrolments/tests/test_validators.py]: yeah that's easier to read as well

File: enrolments/tests/test_validators.py (hunk 2 above)
Comment (Tayef-Shah) [reply enrolments/tests/test_validators.py]: thanks

**Review by brittanylau — APPROVED** [review]
looks great! thank ya thank ya

sorry I slept on this one for a bit 😓 might need to rebase and remake the migrations

**Pull request merged**